volatility3/framework/plugins/windows/mftscan.py

```diff
@@ -0,0 +1,331 @@
+    0x1: "File", # "In Use",
+    0x2: "Directory", # if flag & 0x0002 == 0 this is a regular file
+    0x3: "Directory"
+}
+
+INDEX_ENTRY_FLAGS = {
+    0x1:"Child Node Exists",
+    0x2:"Last entry in list",
+}
+
+
+class MFTScan(interfaces.plugins.PluginInterface):
+    """Scans for MFT FILE objects present in a particular windows memory image."""
+
+    _required_framework_version = (2, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.VersionRequirement(name = 'yarascanner', component = yarascan.YaraScanner,
+                                            version = (2, 0, 0)),
+        ]
+
+    # https://docs.python.org/3/library/struct.html
+    @classmethod
+    def unpack_data(self, mft_record: bytes, offset: int, data_type: str) -> bytes:
+        """Helper to unpack values from the raw mft_record
+        
+        Args:
+            mft_record: 1024 bytes starting from header value as returned by layer read
+            offset: how far in to the record to read
+            data_type: what is the data type to unpack
+
+        Returns:
+            bytes: the unpacked data
+        """
+
+        if data_type == 'unsigned long':
+            return unpack('<L', mft_record[offset:offset+4])[0]
+        elif data_type == 'unsigned short':
+            return unpack('<H', mft_record[offset:offset+2])[0]
+        elif data_type == 'unsigned long long':
+            return unpack('<Q', mft_record[offset:offset+8])[0]
+        elif data_type == 'unsigned char':
+            return unpack('<B', mft_record[offset:offset+1])[0]
+        elif data_type == 'int':
+            return unpack('<I', mft_record[offset:offset+4])[0]
+
+    @classmethod
+    def human_date(self, datetime_object: int) -> str:
+        """Converts a windows epoch to a date time string with a fixed format
+        
+        Args:
+            datetime_object: windows epoch time
+
+        Returns:
+            str: strftime of the windows epoch in UTC
+
+        """
+        dtg = conversion.wintime_to_datetime(datetime_object)
+        return dtg.strftime('%Y-%m-%d %H:%M:%S %z')
+
+    @classmethod
+    def parse_mft_record(self, mft_record: bytes) -> Dict:
+        """Takes an MFT Record and attempts to parse, MFT, SI and FN attributes
+        
+        Args:
+            mft_record: 1024 bytes starting from header value as returned by layer read
+
+        Returns:
+            Dict: a Dictionary that contains the Parse MFT Record
+        """
+        # https://github.com/Invoke-IR/ForensicPosters
+
+        flags = self.unpack_data(mft_record, 22, 'unsigned short')
+        file_type = MFT_FLAGS.get(flags, 'Unknown')
+
+        mft_entry = {
```

Comment:
This is practically a volatility structure, why not define it as one instead of lots of manual `struct.unpack` calls?  We prefer using volatility as the parser wherever possible, and only in instances where it makes a substantial difference to efficiency do we go with using struct.  It would also make it more reusable as other plugins could just load the symbol and use it itself...